client/components/forms/form-radio-with-thumbnail/index.jsx

```diff
@@ -0,0 +1,45 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+
+import FormRadio from 'components/forms/form-radio';
+import FormLabel from 'components/forms/form-label';
+import cssSafeUrl from 'lib/css-safe-url';
+
+const FormRadioWithThumbnail = ( { label, thumbnail, ...otherProps } ) => {
+	const { cssClass, cssColor, imageUrl } = thumbnail;
+	const styles = {
+		backgroundColor: cssColor ? cssColor : '',
+		backgroundImage: imageUrl ? 'url(' + cssSafeUrl( imageUrl ) + ')' : '',
+	};
+
+	return (
+		<div className="form-radio-with-thumbnail">
+			<FormLabel>
+				<div
+					className={ classnames( 'form-radio-with-thumbnail__thumbnail', cssClass ) }
+					style={ styles }
+				/>
+				<FormRadio { ...otherProps } />
+				<span>
+					{ label }
+				</span>
+			</FormLabel>
+		</div>
+	);
+};
+
+FormRadioWithThumbnail.propTypes = {
+	label: PropTypes.string,
+	thumbnail: PropTypes.object,
```

Comment:
Yes, I'd agree that it should be required given that the thumbnail is kind of the main reason this component exists. :)